tag-state/calculate-tag-state.py

```diff
@@ -121,4 +158,122 @@ def semver_sorted_dict(d: dict) -> str:
+            )
+            pull = subprocess.run(["docker", "image", "pull", image_name_by_digest])
+            # tag with the new value
+            if pull.returncode is 0:
+                tag = subprocess.run(
+                    [
+                        "docker",
+                        "image",
+                        "tag",
+                        image_name_by_digest,
+                        f"gcr.io/ironcore-images/{image_name}:{key}",
+                    ]
+                )
+                if tag.returncode is 0:
+                    push = subprocess.run(
+                        ["docker", "push", f"gcr.io/ironcore-images/{image_name}:{key}"]
+                    )
+                    if push.returncode is 0:
+                        value["status"] = SUCCESS
+                        continue
+            value["status"] = FAILED
+    # print what was tagged
+    successfully_tagged = {
+        k: v["digest"] for k, v in tag_state.items() if v["status"] is SUCCESS
+    }
+    failed_tags = {
+        k: v["digest"] for k, v in tag_state.items() if v["status"] is FAILED
+    }
+    needs_manifest = {
+        k: v["digest"] for k, v in tag_state.items() if v["digest"] is NEEDS_MANIFEST
+    }
+    print(pretty_printable_dict(successfully_tagged))
+    print_error(pretty_printable_dict(failed_tags))
+    print(pretty_printable_dict(needs_manifest))
+    # use the tags just pushed to create manifests
+    link_manifest = {}
+    for key in needs_manifest:
+        # we expect no arch tags at this point
+        major, minor, patch = pad_none(key.split("."), 3)
+        # if there's a patch we need to create a manifest for it
+        if patch:
+            gcr_image = f"gcr.io/ironcore-images/{image_name}:{key}"
+            # all our images that need manifests have arm64/amd64 versions
+            create_manifest = subprocess.run(
+                [
+                    "docker",
+                    "manifest",
+                    "create",
+                    gcr_image,
+                    f"{gcr_image}-arm64",
+                    f"{gcr_image}-amd64",
+                ]
+            )
+            if create_manifest.returncode is 0:
+                annotate_manifest_arm = subprocess.run(
+                    [
+                        "docker",
+                        "manifest",
+                        "annotate",
+                        gcr_image,
+                        f"{gcr_image}-arm64",
+                        "--arch",
+                        "arm64",
+                    ]
+                )
+                annotate_manifest_amd = subprocess.run(
+                    [
+                        "docker",
+                        "manifest",
+                        "annotate",
+                        gcr_image,
+                        f"{gcr_image}-amd64",
+                        "--arch",
+                        "amd64",
+                    ]
+                )
```

Comment:
I thought it did the annotate automatically, based on the metadata in the individual images.

Reply:
I haven't seen anything that would make me think that, and looking at the individual images I don't see any arch annotations.

```
λ ~/code/ironcore/buildlog/tag-state/ calculate-tag-state-script docker buildx imagetools inspect gcr.io/ironcore-images/tenant-security-proxy@sha256:4a324c2f6fde40ad2fc30de7f3cbaa97c10bc46a16346ca348bda90de37426ee --raw
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
   "config": {
      "mediaType": "application/vnd.docker.container.image.v1+json",
      "size": 2686,
      "digest": "sha256:a4bccfb049a78cbe5b4e88346650f1b65c9d319326631e6b2bab5fa8d196531f"
   },
   "layers": [
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 2807803,
         "digest": "sha256:91d30c5bc19582de1415b18f1ec5bcbf52a558b62cf6cc201c9669df9f748c22"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 3547379,
         "digest": "sha256:8f1822ef3d7deb98b70e77aa8ee63f9d1e149eb99c2d181d32f1f19007615976"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 5495845,
         "digest": "sha256:258403d433c1ffdcaf842035adf4e329ced556c8956fbfc2aead41554f288bcb"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 5495846,
         "digest": "sha256:acdf53f8785e932039fdafa3294dff0bcb00d36565063ad5327fde4dab93a215"
      }
   ]
}   
```

Reply:
Yeah, I guess you're right. I tried to verify the thing I thought I had seen before, but I must have hallucinated it.